### PR TITLE
exterminator refactor and some changes

### LINKED ISSUE #24330: Terminators can get hungry and thirsty, but can't consume things
Terminators will eventually slow down because of thirst. They might be able to starve to death too?

### PR DESCRIPTION
**Pull request by deltanedas**
## About the PR
make exterminator able to always break out of cuffs since strong

fixes #24330

## Why / Balance
good

## Technical details
renamed everything from Terminator Exterminator

## Media
- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes
exterminator components are in shared and renamed to Exterminator

**Changelog**
:cl:
- add: Exterminators can now break out of cuffs instantly.

### CODE REVIEW

**Comment by IProduceWidgets** [comment]
extremely based

**Comment by IProduceWidgets** [comment]
Probably needs a different motorbike, but that can come later.

If your open to bikesheds the idea seems nice and reusable if the loadout and ftl were able to be set in yml somehow.

**Comment by Kadeo64** [comment]
can't wait for maintainers to 1984 close this because vehicle code is shitty 

jokes aside here are some quick and shitty lever action shotgun sprites 
pump-inhand-left
pump-inhand-right

leveropenbolt
lever

**Comment by K-Dynamic** [comment]
I want this except sometimes you hit your own vehicle when shooting while riding

**Comment by UbaserB** [comment]
Based

**Comment by throge** [comment]
This is fucking insane I love it

**Comment by deltanedas** [comment]
> If your open to bikesheds the idea seems nice and reusable if the loadout and accent were able to be set in yml somehow.

they are but it can only have 1 action

ideally it would be more generic so you could have 5 of them and they all get deleted when you use one but lazy

**Comment by deltanedas** [comment]
> shitty lever action shotgun sprites

99% sure its a pump action shotgun

**Comment by Kadeo64** [comment]
> 99% sure its a pump action shotgun

55943bc369bedd8b53d8d3cf

the one he uses on the bike isn't a pump action

**Comment by IProduceWidgets** [comment]
eh, he uses a few different shotguns, and making this pr run into the weapons freeze seems unnecessary. Could always PR the shotgun later and try to add it to the loadout like I plan to do with a motorbike.

File: Content.Shared/Terminator/Components/TerminatorComponent.cs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+using Robust.Shared.Prototypes;
+
+namespace Content.Shared.Terminator.Components;
+
+/// <summary>
+/// Main exterminator component, handles the target, if any, and adding the curse action.
+/// </summary>
+[RegisterComponent, NetworkedComponent, AutoGenerateComponentState, Access(typeof(SharedTerminatorSystem))]
+public sealed partial class TerminatorComponent : Component
+{
```
Comment (metalgearsloth) [inline Content.Shared/Terminator/Components/TerminatorComponent.cs:14]: What heppened to calling it exterminator to avoid any licensing bs.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]
This went from "cheeky reference" to "disney please dmca us" it looks like to me.

File: Content.Shared/Terminator/Components/TerminatorComponent.cs (hunk 1 above)
Comment (deltanedas) [reply Content.Shared/Terminator/Components/TerminatorComponent.cs:14]: it was Terminator internally already but i can fully rename everything :trollface:

**Comment by MarakCM** [comment]
Maybe add ammo drum (.50 shells)?

**Comment by deltanedas** [comment]
would be unstoppable then, just having a few shells means you can't waste them OR you have to raid security, both need you to be robust

File: Content.Shared/Terminator/Components/TerminatorComponent.cs (hunk 1 above)
Comment (deltanedas) [reply Content.Shared/Terminator/Components/TerminatorComponent.cs]: ok mickey mouse has been fended off everything has been renamed

**Comment by ps3moira** [comment]
I love this

**Comment by VektorZ1** [comment]
It would be cool if you could also control what quote you say, like replacing certain letters with a certain quote.

**Comment by Ilya246** [comment]
> It would be cool if you could also control what quote you say, like replacing certain letters with a certain quote.

levenshtein replacement accent which finds string with least levenshtein distance and says it

**Comment by K-Dynamic** [comment]
do arnie's curse should spawn in a time travelling rebel that guards the target :trollface:

**Comment by deltanedas** [comment]
@metalgearsloth what change i have renamed everything internally to Exterminator

**Comment by metalgearsloth** [comment]
> @metalgearsloth what change i have renamed everything internally to Exterminator

ExterminatorCursed needs removing.

**Comment by deltanedas** [comment]
the entire curse related stuff?

**Comment by deltanedas** [comment]
@metalgearsloth merg

**Comment by SoulFN** [comment]
Can you give exterminator welding protection?

**Comment by SoulFN** [comment]
> they should already be immune to temporary blindness not sure what proper blindness is

When i tried to deconstruct wall with welder i got blindness effect, i was exterminator

**Comment by Golinth** [comment]
Aw, why was the curse removed? I was looking forward to talking in nothing but quotes

**Comment by deltanedas** [comment]
le copyright

**Comment by metalgearsloth** [comment]
Re-merge after newmed stuff ig.

**Pull request closed without merging**